### PR TITLE
Testing tests

### PR DESCRIPTION
**Pull request by Watxesnavo**

### CODE REVIEW

**Comment by Watxesnavo** [comment]
Снимок экрана 2023-12-25 в 11 49 46
Снимок экрана 2023-12-25 в 11 47 31
screenshots of test coverage for both PR Book and ShoppingCart logic

File: src/main/java/org/store/structure/dto/book/CreateBookRequestDto.java [hunk 1]
```diff
@@ -8,12 +11,21 @@
 @Data
 @Accessors(chain = true)
 public class CreateBookRequestDto {
+    @NotNull
+    @Size(min = 1)
     private String title;
+    @NotNull
+    @NotEmpty
```
Comment (olekskov) [inline src/main/java/org/store/structure/dto/book/CreateBookRequestDto.java:18]: `@NotEmpty' rejects null so no need in '@NotNull' when it's present

File: src/main/java/org/store/structure/dto/book/CreateBookRequestDto.java [hunk 2]
```diff
@@ -8,12 +11,21 @@
...
     private String author;
+    @NotNull
+    @NotEmpty
```
Comment (olekskov) [inline src/main/java/org/store/structure/dto/book/CreateBookRequestDto.java:21]: same

File: src/main/java/org/store/structure/dto/order/OrderRequestDto.java [hunk 3]
```diff
@@ -1,10 +1,14 @@
...
+import jakarta.validation.constraints.NotEmpty;
 import jakarta.validation.constraints.NotNull;
 import lombok.Data;
+import lombok.experimental.Accessors;
 
 @Data
+@Accessors(chain = true)
 public class OrderRequestDto {
+    @NotEmpty
     @NotNull
```
Comment (olekskov) [inline src/main/java/org/store/structure/dto/order/OrderRequestDto.java:12]: same

File: src/main/java/org/store/structure/dto/order/OrderStatusUpdateDto.java [hunk 4]
```diff
@@ -1,10 +1,12 @@
 package org.store.structure.dto.order;
 
+import jakarta.validation.constraints.NotEmpty;
 import jakarta.validation.constraints.NotNull;
 import lombok.Data;
 
 @Data
 public class OrderStatusUpdateDto {
+    @NotEmpty
     @NotNull
```
Comment (olekskov) [inline src/main/java/org/store/structure/dto/order/OrderStatusUpdateDto.java:10]: same

File: src/main/java/org/store/structure/service/order/OrderServiceImpl.java [hunk 5]
```diff
@@ -119,7 +118,7 @@ private Set<OrderItem> transformCartToOrderItems(Set<CartItem> cartItems, Order
         return orderItems;
     }
 
-    private Order saveDefaultOrder(UserDetails userDetails) {
+    private Order saveDefaultOrder(User userDetails) {
```
Comment (olekskov) [inline src/main/java/org/store/structure/service/order/OrderServiceImpl.java:121]: User user is more appropriate

File: src/main/java/org/store/structure/service/order/OrderServiceImpl.java [hunk 6]
```diff
@@ -129,8 +128,8 @@ private Order saveDefaultOrder(UserDetails userDetails) {
         return orderRepository.save(order);
     }
 
-    private User getCurrentUser(UserDetails user) {
-        return userRepository.findByEmail(user.getUsername())
+    private User getCurrentUser(User userDetails) {
```
Comment (olekskov) [inline src/main/java/org/store/structure/service/order/OrderServiceImpl.java:131]: same

File: src/test/java/org/store/structure/controller/AuthenticationControllerTest.java [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+    @Test
+    @SneakyThrows
+    void login_failedPasswordValidation_test() {
+        mockMVC.perform(post("/auth/login")
+                        .content(new String(
+                                Files.readAllBytes(
+                                        new File("src/test/resources/request/user"
+                                                + "/user-login-request_invalid-password.json")
+                                                .toPath()))
+                        )
```
Comment (olekskov) [inline src/test/java/org/store/structure/controller/AuthenticationControllerTest.java:134]: extracting content argument to variable will make code more readable

Please recheck similar cases, they are present in most tests

**Review by olekskov — CHANGES_REQUESTED** [review]

File: src/main/java/org/store/structure/service/order/OrderServiceImpl.java (hunk 5 above)
Comment (Watxesnavo) [reply src/main/java/org/store/structure/service/order/OrderServiceImpl.java:121]: @olekskov 
alreadywas
it was there already when i pushed it here.

**Pull request merged**